gmusicapi/protocol/musicmanager.py

```diff
@@ -156,14 +156,14 @@ def get_track_clientid(filepath):
 
             # delete=False is needed because the NamedTemporaryFile
             # can't be opened by name a second time on Windows otherwise.
-            with open(filepath, 'rb') as f, NamedTemporaryFile(suffix=ext, delete=False) as temp:
-                shutil.copy(f.name, temp.name)
+            with NamedTemporaryFile(suffix=ext, delete=False) as temp:
+                shutil.copy(filepath, temp.name)
 
                 audio = mutagen.File(temp.name, easy=True)
                 audio.delete()
                 audio.save()
 
-                m.update(temp.read())
+                m.update(temp.read(65536))
```

Comment:
Oh whoops. I think I hit Ctrl-Z too many times in a hurry and never reread it ^_^